### ISSUE TITLE
[ACRA 2.9] - NullPointerException: VideoPlayer/SoundPlayer

### ISSUE THREAD

**Issue by david-allison-1** [ankidroid/Anki-Android#5911]
https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/311a66d3-8c15-4c1a-b87d-6a31e678a083

```
java.lang.NullPointerException: uriString
at android.net.Uri$StringUri.<init>(Uri.java:475)
at android.net.Uri$StringUri.<init>(Uri.java)
at android.net.Uri.parse(Uri.java:437)
at com.ichi2.libanki.Sound.playSound(Sound.java:3)
at com.ichi2.anki.VideoPlayer.surfaceCreated(VideoPlayer.java:1)
at android.view.SurfaceView.updateWindow(SurfaceView.java:586)
at android.view.SurfaceView$3.onPreDraw(SurfaceView.java:177)
at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:944)
at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2055)
at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1107)
at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6013)
at android.view.Choreographer$CallbackRecord.run(Choreographer.java:858)
at android.view.Choreographer.doCallbacks(Choreographer.java:670)
at android.view.Choreographer.doFrame(Choreographer.java:606)
at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
at android.os.Handler.handleCallback(Handler.java:746)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:148)
at android.app.ActivityThread.main(ActivityThread.java:5421)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

**Comment by david-allison-1** [comment]
This one doesn't seem possible. Maybe if `VideoPlayer.surfaceCreated` is called during `onCreate`

Call (requires path = null to crash):

https://github.com/ankidroid/Anki-Android/blob/547e1c64092acafc197e0d7a6bfe8e3cc0d62aef/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java#L3052-L3057

soundPath is non-null

https://github.com/ankidroid/Anki-Android/blob/0904169051eae3969402ba3f67dba9629748b515/AnkiDroid/src/main/java/com/ichi2/libanki/Sound.java#L285

And soundPath is used in the only call:

https://github.com/ankidroid/Anki-Android/blob/0904169051eae3969402ba3f67dba9629748b515/AnkiDroid/src/main/java/com/ichi2/libanki/Sound.java#L307

I've reproduced construction 10,000 times in RoboElectric and no crash. I'll try on my device later this week.

Activity is `exported:false`, so we control the instantiation unless we're on a heavily modified system.

**Comment by mikehardy** [comment]
We might be getting down to the "ambient noise" crash level :-) - that would be a major achievement but is also something to think about w.r.t. time spent vs people helped. I'm definitely starting to open the flood gates on change again at least (obvious as I plow through Arthur's PRs, but new feature merges will begin again soon). I abhor crashes but I've learned to be pragmatic. At the same time, if you're still enjoying mashing these things down I'm enjoying merging them :-)

**Comment by david-allison-1** [comment]
> We might be getting down to the "ambient noise" crash level :-) - that would be a major achievement but is also something to think about w.r.t. time spent vs people helped. I'm definitely starting to open the flood gates on change again at least (obvious as I plow through Arthur's PRs, but new feature merges will begin again soon). I abhor crashes but I've learned to be pragmatic. At the same time, if you're still enjoying mashing these things down I'm enjoying merging them :-)

Ambient noise-level crashes is great :D, and I'll be really glad to see 2.9.6 out. ACRA is mosty "fixed it" or "non-crash" now, which is wonderful. 

I mostly agree. It's nice to have a list of all crashes on a bugtracker for reference, but I'm not sure GitHub issues is the right place (I don't want things like this taking up anyone's time but mine, and people use the number of issues to judge the health of a project). 

I assumed this would be an easy one to solve, can't catch 'em all though.

In terms of features, I've got 3 planned. I'm guessing 2.10 is failry frozen right now and I should aim for 2.11?

**Comment by mikehardy** [comment]
Yeah - I want to get 2.10 out "nearly as soon as possible", I basically want to work through the stack of Arthur's changes then re-stabilize and ship it so I can zero the translations branch straddle again. I'm going to move some things I have flagged already for 2.10 to 2.11 even and open a 2.12 milestone. In practical terms though that means "aiming for 2.11" is just "code on master" because 2.9.6 is going out today (I've already pushed the official stable version to beta channel) and after it sits a day or so + I merge what I can with Arthur I cut the 2.10 release branch and master is magically 2.11. I won't keep this velocity for ever of course but just as fixing crashes is very motivating, I consider a deep PR backlog a sort of promise to the contributors and I really want to get through them
